README.md

```diff
@@ -1,66 +1,223 @@
-* Download the code by issuing a `go get` command.
+HPE OneView SDK for Terraform can be installed from Source or Docker container installation methods. You can either use a docker container which will have the HPE OneView SDK for terraform installed or perform local installation manually.
+
+### Docker Setup
+The light weight containerized version of the HPE OneView SDK for Terraform is available in the [Docker Store](https://store.docker.com/community/images/hewlettpackardenterprise/hewlettpackardenterprise/hpe-oneview-sdk-for-terraform). The Docker Store image tag consist of two sections: <sdk_version-OV_version>
 
 ```bash
-# Download the source code for terraform-provider-oneview
-# and build the needed binary, by saving it inside $GOPATH/bin
-$ go get -u github.com/HewlettPackard/terraform-provider-oneview
+# Download and store a local copy of oneview-sdk-for-terraform and use it as a Docker Image.
+$ docker pull hewlettpackardenterprise/hpe-oneview-sdk-for-terraform:v1.7.0-OV5.6
+# Run docker commands below given, which  will in turn create a sh session 
+# where you can create files, issue commands and execute the examples.
+$ docker run -it docker pull hewlettpackardenterprise/hpe-oneview-sdk-for-terraform:v1.7.0-OV5.6 /bin/sh
+```
 
-# Copy the binary to have it along the terraform binary
-$ mv $GOPATH/bin/terraform-provider-oneview /usr/local/bin/terraform
+### Local Setup
+
+Local installation requires
+- Installing Go
+```bash 
+$ apt-get install build-essential git wget
+$ wget https://golang.org/dl/go1.15.7.linux-amd64.tar.gz
+
+#unzip and untar the file 
+$ tar -zxvf go1.15.7.linux-amd64.tar.gz
+
+# move it to /usr/local/ and create directory for Go.
+$ mv go/ /usr/local/ 
+$ mkdir ~/go
+```
+```bash 
+# Setting Environment Variable 
+$ export GOROOT=/usr/local/go
+$ export GOPATH=$HOME/go 
+$ export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
 ```
 
+- Installing Terraform
 
+	Install Terraform 0.12.x [from here](https://www.terraform.io/downloads.html) and save it into `/usr/local/bin/terraform` folder (create it if it doesn't exists).
 
-## Using `terraform-provider-oneview` with Docker
+```bash 
+```
 
+- Install Oneview Terraform Provider SDK
+```go
+# Download the source code for terraform-provider-oneview
+# Build the needed binary
+# Get the branch 'Terraform-0.12' terraform-provider-oneview which is supports Terraform v0.12.x.
 
-We also provide a lightweight and easy way to test and execute `terraform-provider-oneview`. The `hewlettpackardenterprise/hpe-oneview-sdk-for-terraform:<tag>` docker image contains an installation of Terraform and our provider you can use by just pulling down the Docker Image:
-Docker Store image tag consist of two sections: <sdk_version-OV_version>
+$ go get github.com/HewlettPackard/terraform-provider-oneview@Terraform-0.12
```

Comment:
this doesn't seems to be working, i followed the exact steps, but no luck, apparently you would need to enable `export GO111MODULE=on`  to enable go get to use versioning.
and fetching from a branch name instead of a release tag is doubtful, i tried no luck. Does that work on your system?
and post performing go get with version, terraform-provider-oneview didn't not populated itself in $GOPATH/src/github... so changing directory as per line 78 didnt worked. 

What did worked was, 
i had to do go get without the version, and then checkout to 0.12 branch ,then did a build.

Reply:
Modified the installation steps